libraries/AP_ExternalAHRS/AP_ExternalAHRS_VectorNav.cpp

```diff
@@ -371,57 +373,70 @@ bool AP_ExternalAHRS_VectorNav::decode(char c)
 }
 
 // decode the most recently consumed term
-// returns true if new sentence has just passed checksum test and is validated
+// returns true if new term is valid
 bool AP_ExternalAHRS_VectorNav::decode_latest_term()
 {
+    // Check the first two terms (In most cases header + reg number) that they match the sent
+    // message. If not, the response is invalid.
     switch (nmea.term_number) {
         case 0:
-            if (strcmp(nmea.term, "VNRRG") != 0) {
+            if (strncmp(nmea.term, message_to_send, nmea.term_offset) != 0) {  // ignore sync byte in comparison
                 return false;
+            } else if (strncmp(nmea.term, "VNERR", nmea.term_offset) == 0) {
+                nmea.error_response = true;  // Message will be printed on next term 
             }
             break;
```

Comment:
This code would be clearer if you just did `return true;` in these `true` cases instead of `break`ing.

Reply:
Complete